### PR TITLE
Update README to reflect behavioural changes in `set`

### LINKED ISSUE #226: Possible need for semver major
With the optimisation changes made in c6a16ed689dea762b2b7c67368ae8c4456d506e7, I've come across an edge case that may require a major release (and possible deprecation of v4.9.0+).

The optimisations included a transition away from using `for (attr in obj)` to `Object.keys(obj)` in order to iterate the object properties. While this is more efficient, and what is necessary nearly all of the time, it changes which properties are iterated over. Essentially, `for…in` loops over all enumerable properties, including those inherited through the prototype chain, while `Object.keys` only returns own attributes that are enumerable.

[See this small example on tonicdev](https://tonicdev.com/dhritzkiv/56a9571b33bb660f00ce789c)

This switch also affects instances of certain constructors like Blob/File, among others. `for...in` loops through those instances' keys (size, name, type, slice), while `Object.keys` does not (at least in Safari). I actually don't know why, but I suspect this has to do with getters/setters and the Browser specific-implementation nature of these instances. ~~[Here's a another small example working with Blobs, on jsbin](http://jsbin.com/lifuhavano/edit?js,console)~~ Since the JSBin example isn't working consistently, I've inlined the code and results:

```js

var blob = new Blob(["hello"]);

console.log(blob.size);//5

console.log("Object.keys", Object.keys(blob)); // "Object.keys" []

var attr;
var attrs = [];

for (attr in blob) {
    attrs.push(attr);
}

console.log("for in", attrs); // "for in" ["size", "slice", "type"]

```

This change actually bit me while working in my ampersand-file-drop-view module which works with Blob/File objects, and `set`ting those blobs onto a State model. (I'm working to make changes in that module to be more explicit about setting the appropriate properties).

Again, these are edge cases, but functionality breaking nonetheless, warranting a major semver release, as annoying as it is. It would also warrant a note in a README.

### PR DESCRIPTION
**Pull request by dhritzkiv**
This PR, along with a major semver change, closes #226

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -622,6 +622,8 @@ Possible options (when using `state.set()`):
 * `silent` {Boolean} - prevents triggering of any change events as a result of the set operation.
 * `unset` {Boolean} - `unset` the attributes keyed in the attributes object instead of setting them.
 
+_Note: when passing an object as the `attributes` argument, only that object's own enumerable properties (i.e. those that can be accessed with `Object.keys(object)` are read and set. This behavious is new as of v5.0.0, as prior version relied on `for...in` to access an object's properties, both owned by that object and those inherited through the prototypal chain._
```
Comment (cdaringe) [inline README.md]: behaviou _r_, :hatching_chick:

Comment (dhritzkiv) [inline README.md]: whoops

**Comment by wraithgar** [comment]
lgtm, shipping this as a major should also coincide w/ a deprecation of *both* 4.9.0 and 4.9.1 imo

**Comment by lukekarrys** [comment]
+1 agree with @wraithgar

**Comment by wraithgar** [comment]
Published as `v5.0.0`

Deprecated `v4.9.0` and `v4.9.1` with message to either pin back at `v4.8.2` or upgrade to `^5.0.0`

**Comment by dhritzkiv** [comment]
👍

**Pull request merged**